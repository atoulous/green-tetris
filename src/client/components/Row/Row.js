import React from 'react';
import Cell from '../Cell/Cell';
import { heightSize } from '../../helpers/constants';

import './Row.scss';

const Row = ({ row }) => {
  let cells = [];
  // Game Grid.
  if (typeof row !== 'number') {
    cells = row.map((e, i) => <Cell key={i} fill={e.fill} color={e.color} />);
  // Player Spectrum.
  } else {
    for (let i = 0; i < heightSize; i++) {
      cells.push(<Cell key={i} fill={row <= i} color="grey" />);
    }
  }
  return <div className="row">{cells}</div>;
};

export default Row;
