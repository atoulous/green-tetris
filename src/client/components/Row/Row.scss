.grid .row {
    display: flex;
    flex-direction: row;
}

.spectrum .row {
    display: flex !important;
    flex-direction: column !important;
    flex: 1;
}