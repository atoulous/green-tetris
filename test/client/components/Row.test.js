import React from 'react';
import renderer from 'react-test-renderer';

import Row from '../../../src/client/components/Row/Row';

it('row with row props number', () => {
  const tree = renderer
    .create(<Row width={20} row={12} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('row with row props array', () => {
  const tree = renderer
    .create(<Row width={20} row={[{ fill: true, color: 'cyan' }, { fill: false, color: 'cyan' }, { fill: true, color: 'red' }, { fill: true, color: 'green' }]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
