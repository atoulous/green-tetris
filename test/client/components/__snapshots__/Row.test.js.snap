// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`row with row props array 1`] = `
<div
  className="row"
>
  <div
    className="cell"
    style={
      Object {
        "backgroundColor": "cyan",
      }
    }
  />
  <div
    className="cell"
    style={Object {}}
  />
  <div
    className="cell"
    style={
      Object {
        "backgroundColor": "red",
      }
    }
  />
  <div
    className="cell"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
</div>
`;

exports[`row with row props number 1`] = `
<div
  className="row"
>
  <div
    className="cell"
    style={Object {}}
  />
  <div
    className="cell"
    style={Object {}}
  />
  <div
    className="cell"
    style={Object {}}
  />
  <div
    className="cell"
    style={Object {}}
  />
  <div
    className="cell"
    style={Object {}}
  />
  <div
    className="cell"
    style={Object {}}
  />
  <div
    className="cell"
    style={Object {}}
  />
  <div
    className="cell"
    style={Object {}}
  />
  <div
    className="cell"
    style={Object {}}
  />
  <div
    className="cell"
    style={Object {}}
  />
  <div
    className="cell"
    style={Object {}}
  />
  <div
    className="cell"
    style={Object {}}
  />
  <div
    className="cell"
    style={
      Object {
        "backgroundColor": "grey",
      }
    }
  />
  <div
    className="cell"
    style={
      Object {
        "backgroundColor": "grey",
      }
    }
  />
  <div
    className="cell"
    style={
      Object {
        "backgroundColor": "grey",
      }
    }
  />
  <div
    className="cell"
    style={
      Object {
        "backgroundColor": "grey",
      }
    }
  />
  <div
    className="cell"
    style={
      Object {
        "backgroundColor": "grey",
      }
    }
  />
  <div
    className="cell"
    style={
      Object {
        "backgroundColor": "grey",
      }
    }
  />
  <div
    className="cell"
    style={
      Object {
        "backgroundColor": "grey",
      }
    }
  />
  <div
    className="cell"
    style={
      Object {
        "backgroundColor": "grey",
      }
    }
  />
  <div
    className="cell"
    style={
      Object {
        "backgroundColor": "grey",
      }
    }
  />
  <div
    className="cell"
    style={
      Object {
        "backgroundColor": "grey",
      }
    }
  />
  <div
    className="cell"
    style={
      Object {
        "backgroundColor": "grey",
      }
    }
  />
  <div
    className="cell"
    style={
      Object {
        "backgroundColor": "grey",
      }
    }
  />
  <div
    className="cell"
    style={
      Object {
        "backgroundColor": "grey",
      }
    }
  />
  <div
    className="cell"
    style={
      Object {
        "backgroundColor": "grey",
      }
    }
  />
  <div
    className="cell"
    style={
      Object {
        "backgroundColor": "grey",
      }
    }
  />
  <div
    className="cell"
    style={
      Object {
        "backgroundColor": "grey",
      }
    }
  />
  <div
    className="cell"
    style={
      Object {
        "backgroundColor": "grey",
      }
    }
  />
  <div
    className="cell"
    style={
      Object {
        "backgroundColor": "grey",
      }
    }
  />
</div>
`;
